### DISCUSSION TITLE
屏蔽平台差异在 Taro 的定位中吗？

### DISCUSSION THREAD

**Discussion by Jeff-Tian** [NervJS/taro#16728] [Q & A]
最近在使用 Swiper 组件时，发现自适应高度 （`adjustHeight` 属性） 在微信小程序中不生效。
![image](https://github.com/user-attachments/assets/8f2c54bb-aac4-4480-979f-78641e074e35)

仔细阅读文档后，发现原来这是**支付宝小程序**的功能。既然支付宝小程序可以做到自适应高度，技术上应该是完全可以做到的，所以这肯定是微信团队在小程序的实现上偷懒了。

不过，Taro 将这些差异透传给开发者真的是最好的做法吗？可以通过一些类似 polyfill 将这些底层的差异屏蔽掉吗？

不清楚这是不是 Taro 的定位。

**我心中的 Taro**：一套代码，可以在所有平台上无差别地运行。

**目前观察到的 Taro**：一套代码，可以打包在所有平台上跑起来，但是跑起来的方式会根据不同的平台有所区别。如果要统一，还得开发者写一些 if / else 来根据不同的平台，写不同的代码。

**Comment by TheKonka** [comment]
这个组件属性的文档就是从各个小程序官方的文档抄过来的，可以说是一个属性的合集，包含所有平台的组件的属性，所以在开发不同平台小程序时，对于某个组件支持哪些属性，还是要参考小程序官方文档的。
有一些差异 Taro 会抹平，但是像这种组件的属性，如果小程序平台自己不支持，Taro 也很难支持吧

**Reply by Jeff-Tian** [reply]
对，比较难以支持。如果是在 Taro 的定位中，我想尝试提个 PR 在微信小程序中对这个属性进行 polyfill。
